File: 349-intersection-of-two-arrays/intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        # nums1.sort()
        # nums2.sort()
        # ans=[]
        # n=len(nums1)
        # m=len(nums2)
        # i=0
        # j=0
        # while i<n and j<m:
        #     if nums1[i]==nums2[j] and (not ans or ans[-1]!=nums1[i]):
        #         ans.append(nums1[i])
        #         i+=1
        #         j+=1
        #     elif nums1[i]<nums2[j]:
        #         i+=1
        #     else:
        #         j+=1
        # return ans


        
        ans1=set(nums1)
        ans2=set(nums2)
        ans=list(ans1.intersection(ans2))
        return ans

File: 349-intersection-of-two-arrays/intersection-of-two-arrays.cpp
// class Solution {
// public:
//     vector<int> intersection(vector<int>& a, vector<int>& b) {
//     unordered_set<int> s(a.begin(), a.end());
//     vector<int> ans;
//     for (int x : b) {
//         if (s.count(x)) {
//             ans.push_back(x);
//             s.erase(x);
//         }
//     }
//     return ans;
//     }
// };



class Solution {
public:
    vector<int> intersection(vector<int>& a, vector<int>& b) {
        vector<int>ans;
        unordered_map<int,int>mp;
        for(int i=0;i<a.size();i++){
            if(mp.find(a[i])==mp.end()){
                mp[a[i]]++;
            }
        }
        for(int i=0;i<b.size();i++){
            if(mp.find(b[i])!=mp.end()){
                ans.push_back(b[i]);
                mp.erase(b[i]);
            }
        }
        return ans;
    }
};